packages/components/src/dropdown-menu/test/index.js

```diff
@@ -49,62 +57,46 @@ describe( 'DropdownMenu', () => {
 				onClick: jest.fn(),
 			},
 		];
-	} );
 
-	describe( 'basic rendering', () => {
-		it( 'should render a null element when neither controls nor children are assigned', () => {
-			const { container } = render( <DropdownMenu /> );
+		render( <DropdownMenu controls={ controls } /> );
 
-			expect( container.firstChild ).toBeNull();
-		} );
+		// Move focus on the toggle button
+		await user.tab();
 
-		it( 'should render a null element when controls are empty and children is not specified', () => {
-			const { container } = render( <DropdownMenu controls={ [] } /> );
+		await user.keyboard( '[ArrowDown]' );
 
-			expect( container.firstChild ).toBeNull();
-		} );
+		await waitFor( () =>
+			expect( screen.getByRole( 'menu' ) ).toBeVisible()
+		);
 
-		it( 'should open menu on arrow down (controls)', () => {
-			const {
-				container: { firstChild: dropdownMenuContainer },
-			} = render( <DropdownMenu controls={ controls } /> );
-
-			const button = getMenuToggleButton( dropdownMenuContainer );
-			button.focus();
-			fireEvent.keyDown( button, {
-				keyCode: DOWN,
-				preventDefault: () => {},
-			} );
-			const menu = getNavigableMenu( dropdownMenuContainer );
-			expect( menu ).toBeTruthy();
-
-			expect(
-				dropdownMenuContainer.querySelectorAll(
-					'.components-dropdown-menu__menu-item'
-				)
-			).toHaveLength( controls.length );
+		expect( screen.getAllByRole( 'menuitem' ) ).toHaveLength(
+			controls.length
+		);
+	} );
+
+	it( 'should open menu when pressing arrow down on the toggle and the children prop is used to define menu items', async () => {
+		const user = userEvent.setup( {
+			advanceTimers: jest.advanceTimersByTime,
 		} );
 
-		it( 'should open menu on arrow down (children)', () => {
-			const {
-				container: { firstChild: dropdownMenuContainer },
-			} = render( <DropdownMenu children={ children } /> );
+		render(
+			<DropdownMenu
+				children={ ( { onClose } ) => <MenuItem onClick={ onClose } /> }
+			/>
+		);
 
-			const button = getMenuToggleButton( dropdownMenuContainer );
-			button.focus();
-			fireEvent.keyDown( button, {
-				keyCode: DOWN,
-				preventDefault: () => {},
-			} );
+		const button = screen.getByRole( 'button' );
+		button.focus();
 
-			expect( getNavigableMenu( dropdownMenuContainer ) ).toBeTruthy();
+		await user.keyboard( '[ArrowDown]' );
 
-			const menuItem = dropdownMenuContainer.querySelector(
-				'.components-menu-item__button'
-			);
-			fireEvent.click( menuItem );
+		await waitFor( () =>
+			expect( screen.getByRole( 'menu' ) ).toBeVisible()
+		);
 
-			expect( getNavigableMenu( dropdownMenuContainer ) ).toBeNull();
-		} );
+		// Clicking the menu item will close the dropdown menu
+		await user.click( screen.getByRole( 'menuitem' ) );
```

Comment:
Noting that we no longer care about whether the `menuitem` is within the above `menu` - is that a problem? In the previous version of the tests, we were looking for menuitems only within the menu. This is achievable with `{within} from '@testing-library/dom'` for the protocol.

Reply:
Great call, I pushed https://github.com/WordPress/gutenberg/pull/43439/commits/a50dc67a6d60bf5349abe9d4298d030c2077b6df — let me know if this is what you had in mind!

Reply:
Exactly what I meant - thanks 👍